### ISSUE TITLE
Accessing destroyed object in shell/gmenu.js:514

### ISSUE THREAD

**Issue by Reinis** [andyholmes/gnome-shell-extension-gsconnect#487]
**Describe the bug**
Accessing destroyed object in `shell/gmenu.js`. There is a stack trace in system log every time it looses connection with the phone. And enabling debug log just generated an additional one.

**Debug output**
```
GSConnect Version: 20
GSConnect Install: system
GJS: 15205
XDG_SESSION_TYPE: x11
GDMSESSION: gnome-xorg
--------------------------------------------------------------------------------
-- Logs begin at Wed 2018-10-10 10:35:02 EEST, end at Fri 2019-03-15 15:12:13 EET. --
mar 15 14:53:40 gnome-shell[2684]: Object Gio.Settings (0x5634cdba1970), has been already deallocated - impossible to access it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: == Stack trace for context 0x5634cb9a04d0 ==
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #0 0x5634cbd608a0 i   resource:///org/gnome/shell/ui/messageTray.js:235 (0x7fd330299b38 @ 22)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #1 0x7ffc9b004eb0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #2 0x5634cbd60800 i   resource:///org/gnome/shell/ui/messageTray.js:812 (0x7fd33029eb38 @ 28)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #3 0x7ffc9b005a50 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #4 0x5634cbd60760 i   resource:///org/gnome/shell/ui/windowAttentionHandler.js:100 (0x7fd31c4456f8 @ 42)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #5 0x7ffc9b0065f0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #6 0x5634cbd606e8 i   resource:///org/gnome/shell/ui/windowAttentionHandler.js:44 (0x7fd31c445230 @ 17)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #7 0x7ffc9b0071b0 I   resource:///org/gnome/gjs/modules/signals.js:128 (0x7fd3305cff78 @ 386)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #8 0x5634cbd60660 i   resource:///org/gnome/shell/ui/messageTray.js:479 (0x7fd33029c670 @ 22)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #9 0x7ffc9b007d50 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #10 0x5634cbd605e0 i   resource:///org/gnome/shell/ui/calendar.js:801 (0x7fd3302a9a28 @ 22)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #11 0x7ffc9b008900 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #12 0x7ffc9b0089d0 b   self-hosted:918 (0x7fd3305ebf78 @ 394)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: == Stack trace for context 0x5634cb9a04d0 ==
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #0 0x5634cbd608a0 i   resource:///org/gnome/shell/ui/messageTray.js:236 (0x7fd330299b38 @ 42)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #1 0x7ffc9b004eb0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #2 0x5634cbd60800 i   resource:///org/gnome/shell/ui/messageTray.js:812 (0x7fd33029eb38 @ 28)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #3 0x7ffc9b005a50 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #4 0x5634cbd60760 i   resource:///org/gnome/shell/ui/windowAttentionHandler.js:100 (0x7fd31c4456f8 @ 42)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #5 0x7ffc9b0065f0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #6 0x5634cbd606e8 i   resource:///org/gnome/shell/ui/windowAttentionHandler.js:44 (0x7fd31c445230 @ 17)
mar 15 14:53:40 gnome-shell[2684]: g_object_run_dispose: assertion 'G_IS_OBJECT (object)' failed
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #7 0x7ffc9b0071b0 I   resource:///org/gnome/gjs/modules/signals.js:128 (0x7fd3305cff78 @ 386)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #8 0x5634cbd60660 i   resource:///org/gnome/shell/ui/messageTray.js:479 (0x7fd33029c670 @ 22)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #9 0x7ffc9b007d50 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #10 0x5634cbd605e0 i   resource:///org/gnome/shell/ui/calendar.js:801 (0x7fd3302a9a28 @ 22)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #11 0x7ffc9b008900 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)
mar 15 14:53:40 org.gnome.Shell.desktop[2684]: #12 0x7ffc9b0089d0 b   self-hosted:918 (0x7fd3305ebf78 @ 394)
mar 15 14:53:40 gnome-shell[2684]: Object Gio.Settings (0x5634cf8fe700), has been already deallocated - impossible to access it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs
mar 15 14:53:40 gnome-shell[2684]: g_object_run_dispose: assertion 'G_IS_OBJECT (object)' failed
mar 15 14:55:07 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552654507482,
                                                               "type": "kdeconnect.clipboard",
                                                               "body": {
                                                                 "content": "mar 15 14:53:40 rd-oc gnome-shell[2684]: Object Gio.Settings (0x5634cdba1970), has been already deallocated - impossible to access it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: == Stack trace for context 0x5634cb9a04d0 ==\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #0 0x5634cbd608a0 i   resource:///org/gnome/shell/ui/messageTray.js:235 (0x7fd330299b38 @ 22)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #1 0x7ffc9b004eb0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #2 0x5634cbd60800 i   resource:///org/gnome/shell/ui/messageTray.js:812 (0x7fd33029eb38 @ 28)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #3 0x7ffc9b005a50 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #4 0x5634cbd60760 i   resource:///org/gnome/shell/ui/windowAttentionHandler.js:100 (0x7fd31c4456f8 @ 42)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #5 0x7ffc9b0065f0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #6 0x5634cbd606e8 i   resource:///org/gnome/shell/ui/windowAttentionHandler.js:44 (0x7fd31c445230 @ 17)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #7 0x7ffc9b0071b0 I   resource:///org/gnome/gjs/modules/signals.js:128 (0x7fd3305cff78 @ 386)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #8 0x5634cbd60660 i   resource:///org/gnome/shell/ui/messageTray.js:479 (0x7fd33029c670 @ 22)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #9 0x7ffc9b007d50 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #10 0x5634cbd605e0 i   resource:///org/gnome/shell/ui/calendar.js:801 (0x7fd3302a9a28 @ 22)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #11 0x7ffc9b008900 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #12 0x7ffc9b0089d0 b   self-hosted:918 (0x7fd3305ebf78 @ 394)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: == Stack trace for context 0x5634cb9a04d0 ==\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #0 0x5634cbd608a0 i   resource:///org/gnome/shell/ui/messageTray.js:236 (0x7fd330299b38 @ 42)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #1 0x7ffc9b004eb0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #2 0x5634cbd60800 i   resource:///org/gnome/shell/ui/messageTray.js:812 (0x7fd33029eb38 @ 28)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #3 0x7ffc9b005a50 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #4 0x5634cbd60760 i   resource:///org/gnome/shell/ui/windowAttentionHandler.js:100 (0x7fd31c4456f8 @ 42)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #5 0x7ffc9b0065f0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #6 0x5634cbd606e8 i   resource:///org/gnome/shell/ui/windowAttentionHandler.js:44 (0x7fd31c445230 @ 17)\nmar 15 14:53:40 rd-oc gnome-shell[2684]: g_object_run_dispose: assertion 'G_IS_OBJECT (object)' failed\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #7 0x7ffc9b0071b0 I   resource:///org/gnome/gjs/modules/signals.js:128 (0x7fd3305cff78 @ 386)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #8 0x5634cbd60660 i   resource:///org/gnome/shell/ui/messageTray.js:479 (0x7fd33029c670 @ 22)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #9 0x7ffc9b007d50 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #10 0x5634cbd605e0 i   resource:///org/gnome/shell/ui/calendar.js:801 (0x7fd3302a9a28 @ 22)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #11 0x7ffc9b008900 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7fd3305b5de0 @ 71)\nmar 15 14:53:40 rd-oc org.gnome.Shell.desktop[2684]: #12 0x7ffc9b0089d0 b   self-hosted:918 (0x7fd3305ebf78 @ 394)\nmar 15 14:53:40 rd-oc gnome-shell[2684]: Object Gio.Settings (0x5634cf8fe700), has been already deallocated - impossible to access it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs\nmar 15 14:53:40 rd-oc gnome-shell[2684]: g_object_run_dispose: assertion 'G_IS_OBJECT (object)' failed\n"
                                                               }
                                                             }
mar 15 15:01:01 CROND[23320]: (root) CMD (run-parts /etc/cron.hourly)
mar 15 15:01:22 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:receive/<:177]: Sony: {
                                                               "id": 1552653504594,
                                                               "type": "kdeconnect.battery",
                                                               "body": {
                                                                 "currentCharge": 75,
                                                                 "isCharging": false,
                                                                 "thresholdEvent": 0
                                                               }
                                                             }
mar 15 15:01:23 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/lan.js:_onIdentity:238]: {
                                                               "id": 1552654883829,
                                                               "type": "kdeconnect.identity",
                                                               "body": {
                                                                 "deviceId": "c032d85b9eb38e1b",
                                                                 "deviceName": "Sony",
                                                                 "protocolVersion": 7,
                                                                 "deviceType": "phone",
                                                                 "incomingCapabilities": [
                                                                   "kdeconnect.sms.request_conversations",
                                                                   "kdeconnect.telephony.request_mute",
                                                                   "kdeconnect.telephony.request",
                                                                   "kdeconnect.mpris",
                                                                   "kdeconnect.notification",
                                                                   "kdeconnect.sms.request_conversation",
                                                                   "kdeconnect.findmyphone.request",
                                                                   "kdeconnect.ping",
                                                                   "kdeconnect.systemvolume",
                                                                   "kdeconnect.notification.reply",
                                                                   "kdeconnect.share.request",
                                                                   "kdeconnect.sftp.request",
                                                                   "kdeconnect.notification.request",
                                                                   "kdeconnect.mousepad.request",
                                                                   "kdeconnect.contacts.request_vcards_by_uid",
                                                                   "kdeconnect.sms.request",
                                                                   "kdeconnect.runcommand",
                                                                   "kdeconnect.battery.request",
                                                                   "kdeconnect.clipboard",
                                                                   "kdeconnect.contacts.request_all_uids_timestamps"
                                                                 ],
                                                                 "outgoingCapabilities": [
                                                                   "kdeconnect.sms.messages",
                                                                   "kdeconnect.telephony",
                                                                   "kdeconnect.notification",
                                                                   "kdeconnect.contacts.response_uids_timestamps",
                                                                   "kdeconnect.findmyphone.request",
                                                                   "kdeconnect.ping",
                                                                   "kdeconnect.mousepad.keyboardstate",
                                                                   "kdeconnect.share.request",
                                                                   "kdeconnect.contacts.response_vcards",
                                                                   "kdeconnect.notification.request",
                                                                   "kdeconnect.mousepad.echo",
                                                                   "kdeconnect.mousepad.request",
                                                                   "kdeconnect.sftp",
                                                                   "kdeconnect.runcommand.request",
                                                                   "kdeconnect.mpris.request",
                                                                   "kdeconnect.systemvolume.request",
                                                                   "kdeconnect.battery",
                                                                   "kdeconnect.clipboard"
                                                                 ],
                                                                 "tcpPort": 1716,
                                                                 "tcpHost": "192.168.1.4"
                                                               }
                                                             }
mar 15 15:01:23 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/lan.js:_onIdentity:262]: "already connected"
mar 15 15:08:42 gnome-shell[2684]: Failed to get Yr.no forecast data: 2 Kļūda, sameklējot “api.met.no” — Temporary failure in name resolution
mar 15 15:09:42 gnome-shell[2684]: Failed to get Yr.no forecast data: 2 Kļūda, sameklējot “api.met.no” — Temporary failure in name resolution
mar 15 15:10:32 gnome-shell[2684]: Failed to get Yr.no forecast data: 2 Kļūda, sameklējot “api.met.no” — Temporary failure in name resolution
mar 15 15:11:32 gnome-shell[2684]: Failed to get Yr.no forecast data: 2 Kļūda, sameklējot “api.met.no” — Temporary failure in name resolution
mar 15 15:12:05 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:receive/<:165]: Sony: Kļūda, saņemot datus — Connection reset by peer
                                                             receive/<@/usr/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/protocol/core.js:165:32
                                                             @/usr/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:744:2
mar 15 15:12:05 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/lan.js:close:570]: "Channel (tcp)"
mar 15 15:12:05 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/device.js:_setDisconnected:337]: "Disconnected from Sony (c032d85b9eb38e1b)"
mar 15 15:12:05 gnome-shell[2684]: Object .GSConnectShellIconButton (0x5634cd7a3da0), has been already finalized. Impossible to get any property from it.
mar 15 15:12:05 org.gnome.Shell.desktop[2684]: == Stack trace for context 0x5634cb9a04d0 ==
mar 15 15:12:05 org.gnome.Shell.desktop[2684]: #0 0x5634cbd605e0 i   /usr/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/shell/gmenu.js:514 (0x7fd2ed42f2b8 @ 124)
mar 15 15:12:05 org.gnome.Shell.desktop[2684]: #1 0x7ffc9b008df0 b   self-hosted:922 (0x7fd3305ebf78 @ 469)
mar 15 15:12:06 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/lan.js:_onIdentity:238]: {
                                                               "id": 1552655526549,
                                                               "type": "kdeconnect.identity",
                                                               "body": {
                                                                 "deviceId": "c032d85b9eb38e1b",
                                                                 "deviceName": "Sony",
                                                                 "protocolVersion": 7,
                                                                 "deviceType": "phone",
                                                                 "incomingCapabilities": [
                                                                   "kdeconnect.sms.request_conversations",
                                                                   "kdeconnect.telephony.request_mute",
                                                                   "kdeconnect.telephony.request",
                                                                   "kdeconnect.mpris",
                                                                   "kdeconnect.notification",
                                                                   "kdeconnect.sms.request_conversation",
                                                                   "kdeconnect.findmyphone.request",
                                                                   "kdeconnect.ping",
                                                                   "kdeconnect.systemvolume",
                                                                   "kdeconnect.notification.reply",
                                                                   "kdeconnect.share.request",
                                                                   "kdeconnect.sftp.request",
                                                                   "kdeconnect.notification.request",
                                                                   "kdeconnect.mousepad.request",
                                                                   "kdeconnect.contacts.request_vcards_by_uid",
                                                                   "kdeconnect.sms.request",
                                                                   "kdeconnect.runcommand",
                                                                   "kdeconnect.battery.request",
                                                                   "kdeconnect.clipboard",
                                                                   "kdeconnect.contacts.request_all_uids_timestamps"
                                                                 ],
                                                                 "outgoingCapabilities": [
                                                                   "kdeconnect.sms.messages",
                                                                   "kdeconnect.telephony",
                                                                   "kdeconnect.notification",
                                                                   "kdeconnect.contacts.response_uids_timestamps",
                                                                   "kdeconnect.findmyphone.request",
                                                                   "kdeconnect.ping",
                                                                   "kdeconnect.mousepad.keyboardstate",
                                                                   "kdeconnect.share.request",
                                                                   "kdeconnect.contacts.response_vcards",
                                                                   "kdeconnect.notification.request",
                                                                   "kdeconnect.mousepad.echo",
                                                                   "kdeconnect.mousepad.request",
                                                                   "kdeconnect.sftp",
                                                                   "kdeconnect.runcommand.request",
                                                                   "kdeconnect.mpris.request",
                                                                   "kdeconnect.systemvolume.request",
                                                                   "kdeconnect.battery",
                                                                   "kdeconnect.clipboard"
                                                                 ],
                                                                 "tcpPort": 1716,
                                                                 "tcpHost": "192.168.1.4"
                                                               }
                                                             }
mar 15 15:12:07 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/lan.js:_sendIdent/<:516]: "sending identity"
mar 15 15:12:07 gjs[32562]: JS WARNING: [/usr/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/protocol/lan.js 518]: reference to undefined property "GSocketOutputStream"
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/device.js:_setConnected:323]: "Connected to Sony (c032d85b9eb38e1b)"
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533486,
                                                               "type": "kdeconnect.battery.request",
                                                               "body": {
                                                                 "request": true
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533492,
                                                               "type": "kdeconnect.contacts.request_all_uids_timestamps",
                                                               "body": {}
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533494,
                                                               "type": "kdeconnect.mousepad.keyboardstate",
                                                               "body": {
                                                                 "state": true
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533494,
                                                               "type": "kdeconnect.mpris",
                                                               "body": {
                                                                 "playerList": [],
                                                                 "supportAlbumArtPayload": true
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533496,
                                                               "type": "kdeconnect.notification.request",
                                                               "body": {
                                                                 "request": true
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533496,
                                                               "type": "kdeconnect.runcommand",
                                                               "body": {
                                                                 "commandList": {}
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533496,
                                                               "type": "kdeconnect.runcommand.request",
                                                               "body": {
                                                                 "requestCommandList": true
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533498,
                                                               "type": "kdeconnect.sftp.request",
                                                               "body": {
                                                                 "startBrowsing": true
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533498,
                                                               "type": "kdeconnect.sms.request_conversations",
                                                               "body": {}
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533500,
                                                               "type": "kdeconnect.systemvolume",
                                                               "body": {
                                                                 "sinkList": [
                                                                   {
                                                                     "name": "alsa_output.pci-0000_00_1b.0.analog-stereo",
                                                                     "description": "Speakers (Built-in Audio Analog Stereo)",
                                                                     "muted": false,
                                                                     "volume": 26831,
                                                                     "maxVolume": 65536
                                                                   }
                                                                 ]
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:receive/<:177]: Sony: {
                                                               "id": 1552655534132,
                                                               "type": "kdeconnect.battery",
                                                               "body": {
                                                                 "currentCharge": 75,
                                                                 "isCharging": false,
                                                                 "thresholdEvent": 0
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:receive/<:177]: Sony: {
                                                               "id": 1552655534142,
                                                               "type": "kdeconnect.mpris.request",
                                                               "body": {
                                                                 "requestPlayerList": true
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533560,
                                                               "type": "kdeconnect.mpris",
                                                               "body": {
                                                                 "playerList": [],
                                                                 "supportAlbumArtPayload": true
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:receive/<:177]: Sony: {
                                                               "id": 1552655534145,
                                                               "type": "kdeconnect.runcommand.request",
                                                               "body": {
                                                                 "requestCommandList": true
                                                               }
                                                             }
mar 15 15:12:13 org.gnome.Shell.Extensions.GSConnect[32562]: [/service/protocol/core.js:send:220]: Sony: {
                                                               "id": 1552655533564,
                                                               "type": "kdeconnect.runcommand",
                                                               "body": {
                                                                 "commandList": {}
                                                               }
                                                             }
```
**System Details:**
 - GSConnect version: de325151271bdea437c4188605dd936f26199de2
   - Installed from: live ebuild from overlay [rei4dan](https://github.com/Reinis/rei4dan)
 - GNOME/Shell version: 3.28.3
 - Distro/Release: Gentoo Base System release 2.6

**Comment by andyholmes** [comment]
Thanks for reporting, I'll have a look into this when I get a bit of time.

**Comment by ferdnyc** [comment]
@andyholmes 

This code hasn't been touched in _months_ so it's weird that this would only be coming up now, but... the `IconBox` method `_onItemsChanged` is accessing `button.action_name` _after_ calling `destroy()` on the `button`, which seems like it probably shouldn't work?

https://github.com/andyholmes/gnome-shell-extension-gsconnect/blob/40335f7009a404cbff9e88906c788ea1cb67ee6d/src/shell/gmenu.js#L507-L516

**Comment by andyholmes** [comment]
Should be fixed in 6c8ab5d. Thanks for reporting.
